packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1977,6 +1985,59 @@ export class ContainerRuntime
 		this.removeAllListeners();
 	}
 
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#IContainerRuntimeBase.isSnapshotFetchRequired}
+	 */
+	public isSnapshotFetchRequired(pathParts: string[]): boolean {
+		validatePrecondition(
```

Comment:
Why not use asserts? This is not public API (or rather - it should not be, it should be internal). And thus, I'd rather optimize bundle size (not have strings in bundle).

The boundary between what is "public" vs. "internal" is not very clear, but I think we can assume that that implementing IFluidDataStoreRuntime is not something we support publicly (yet, or ever).

Reply:
Same for UsageError() below. I do not think that having path in telemetry would be useful, as we do not have access to a file. It would be useful if we completely screwed up with its generation (like it has 3 elements instead of 2 - something like that), but otherwise it will not give us much.

Also - this could be EUPI, so it has to have appropriate tag (tagCodeArtifacts?), if you were to log it.

Reply:
Replaced with asserts.